cmd/kubeadm/app/phases/markcontrolplane/markcontrolplane.go

```diff
@@ -18,19 +18,27 @@ package markcontrolplane
 
 import (
 	"fmt"
+	"strings"
 
 	"k8s.io/api/core/v1"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
 	"k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient"
 )
 
-// MarkControlPlane taints the control-plane and sets the control-plane label
-func MarkControlPlane(client clientset.Interface, controlPlaneName string, taints []v1.Taint) error {
+const deprecatedSuffix = "(deprecated)"
 
+var labelsToAdd = []string{
 	// TODO: https://github.com/kubernetes/kubeadm/issues/2200
-	fmt.Printf("[mark-control-plane] Marking the node %s as control-plane by adding the labels \"%s=''\" and \"%s='' (deprecated)\"\n",
-		controlPlaneName, constants.LabelNodeRoleOldControlPlane, constants.LabelNodeRoleControlPlane)
+	constants.LabelNodeRoleOldControlPlane + deprecatedSuffix,
```

Comment:
might be I'm missing something, but why are we adding a deprecated suffix for removing it down in `markControlPlaneNode`?

Reply:
If the goal is to make the message in MarkControlPlane to report (deprecated), I would prefer to have something, might be an `AddDeprecationNotice` func,   that manipulates `labelsToAdd` before calling fmt.Printf instead of having this UX concern spreading around the code base.

Reply:
that's the other way of doing it.
will amend later today.